hwy/contrib/random/random.h

```diff
@@ -0,0 +1,219 @@
+
+#if defined(HIGHWAY_HWY_CONTRIB_RANDOM_RANDOM_H_) == \
+    defined(HWY_TARGET_TOGGLE)  // NOLINT
+#ifdef HIGHWAY_HWY_CONTRIB_RANDOM_RANDOM_H_
+#undef HIGHWAY_HWY_CONTRIB_RANDOM_RANDOM_H_
+#else
+#define HIGHWAY_HWY_CONTRIB_RANDOM_RANDOM_H_
+#endif
+
+#include <hwy/aligned_allocator.h>
+#include <hwy/highway.h>
+
+#include <array>
+#include <cstdint>
+#include <limits>
+
+HWY_BEFORE_NAMESPACE();  // required if not using HWY_ATTR
+
+namespace hwy {
+
+namespace HWY_NAMESPACE {  // required: unique per target
+
+namespace internal {
+class SplitMix64 {
+ public:
+  constexpr SplitMix64(const std::uint64_t state) noexcept : m_state(state) {}
+
+  constexpr std::uint64_t operator()() {
+    std::uint64_t z = (m_state += 0x9e3779b97f4a7c15);
+    z = (z ^ (z >> 30)) * 0xbf58476d1ce4e5b9;
+    z = (z ^ (z >> 27)) * 0x94d049bb133111eb;
+    return z ^ (z >> 31);
+  }
+
+ private:
+  std::uint64_t m_state;
+};
+
+class Xoshiro {
+ public:
+  constexpr explicit Xoshiro(const std::uint64_t seed) noexcept : m_state{} {
```

Comment:
We aim to build with C++11. We have HWY_CXX14_CONSTEXPR in base.h, but I think C++14 isn't even enough here. Would you like to add a CXX17 version?